# HG changeset patch
# User peytoia
# Date 1316069726 -32400
#      Thu Sep 15 15:55:26 2011 +0900
# Node ID 48eafb721da1c1412935dc629ff8e1bd311c4be7
# Parent  d17cd240b3848fb1e831a58b36fc0ecba6b0cb3a
7090844, PR2298: Support a timezone whose offset is changed more than once in the future
Reviewed-by: okutsu

diff -Nru openjdk.orig/jdk/make/tools/src/build/tools/javazic/Mappings.java openjdk/jdk/make/tools/src/build/tools/javazic/Mappings.java
--- openjdk.orig/jdk/make/tools/src/build/tools/javazic/Mappings.java	2015-07-22 02:43:26.831650496 +0100
+++ openjdk/jdk/make/tools/src/build/tools/javazic/Mappings.java	2015-07-22 02:45:35.313410942 +0100
@@ -72,17 +72,20 @@
             Zone zone = zones.get(zoneName);
             String zonename = zone.getName();
             int rawOffset = zone.get(zone.size()-1).getGmtOffset();
+
             // If the GMT offset of this Zone will change in some
             // future time, this Zone is added to the exclude list.
-            if (zone.size() > 1) {
-                ZoneRec zrec = zone.get(zone.size()-2);
+            boolean isExcluded = false;
+            for (int i = 0; i < zone.size(); i++) {
+                ZoneRec zrec = zone.get(i);
                 if ((zrec.getGmtOffset() != rawOffset)
                     && (zrec.getUntilTime(0) > Time.getCurrentTime())) {
                     if (excludeList == null) {
                         excludeList = new ArrayList<String>();
                     }
                     excludeList.add(zone.getName());
-                    continue;
+                    isExcluded = true;
+                    break;
                 }
             }
 
@@ -98,9 +101,11 @@
                 rawOffsetsIndex.add(i, rawOffset);
 
                 Set<String> perRawOffset = new TreeSet<String>();
-                perRawOffset.add(zonename);
+                if (!isExcluded) {
+                    perRawOffset.add(zonename);
+                }
                 rawOffsetsIndexTable.add(i, perRawOffset);
-            } else {
+            } else if (!isExcluded) {
                 int i = rawOffsetsIndex.indexOf(new Integer(rawOffset));
                 Set<String> perRawOffset = rawOffsetsIndexTable.get(i);
                 perRawOffset.add(zonename);
